### PR TITLE
Edited math library and including extensions process

### PR DESCRIPTION
**Pull request by CornerPin**
- Fixed some mistakes with registering functions, like defining atan function as math.asin
- Extensions are loading by finding them in extensions directory
- Added few math functions

### CODE REVIEW

File: lua/expression3/extensions/math.lua [hunk 1]
```diff
@@ -17,13 +17,32 @@ local extension = EXPR_LIB.RegisterExtenstion("math");
 -- TODO: Add math.pi as constant
```
Comment (CaptainPRICE) [inline lua/expression3/extensions/math.lua:17]: Get rid of the comment?

File: lua/expression3/extensions/math.lua [hunk 2]
```diff
@@ -17,13 +17,32 @@ local extension = EXPR_LIB.RegisterExtenstion("math");
...
+local tan  = math.tan;
+local atan = math.atan;
 
 
+local function cot( radians )
+	return 1 / tan( radians );
+end
+
+local function acot( radians )
+	return PI/2 - atan( radians );
```
Comment (CaptainPRICE) [inline lua/expression3/extensions/math.lua]: How about localizing `PI/2`?

Comment (CaptainPRICE) [reply lua/expression3/extensions/math.lua]: Division is slow when it comes to math. So, either localize it or use `*0.5` instead.

Comment (bigdogmat) [reply lua/expression3/extensions/math.lua]: The difference would be soo negligible, also considering it's local I wouldn't be surprised if it was compiled out anyway

File: lua/expression3/extensions/math.lua (hunk 1 above)
Comment (CornerPin) [reply lua/expression3/extensions/math.lua:17]: I think it's about adding pi as constant to E3 math library, but E3 doesn't supports constants for now as i know

**Pull request merged**